src/System/Tensor2_3D.cpp

```diff
@@ -596,4 +596,26 @@ Tensor2_3D Ten2::gen_zero()
       0.0, 0.0, 0.0 );
 }
 
+Tensor2_3D Ten2::exp( const Tensor2_3D &input )
+{
+  double nn = 0.0;
+  double nn_fac = 1.0;
+
+  Tensor2_3D input_pow = Ten2::gen_id();
+  Tensor2_3D input_exp = Ten2::gen_id();
+
+  do
+  {
+    nn += 1.0;
+    nn_fac *= nn;
+
+    const Tensor2_3D temp = input_pow * input;
+    input_pow = temp;
```

Comment:
another question: do we really need a temp tensor here. Can we just do input_pow = input_pow * input?

Reply:
I assume our * operator allows us to do so. @guanjiashen can you confirm?

Reply:
My tests tell me we can use input_pow = input_pow  * input. In fact, the * operator only returns a new Tensor2_3D data.

Reply:
Yes, I have deleted the temp tensor.